integration-cli/docker_cli_run_test.go

```diff
@@ -4058,3 +4058,64 @@ func (s *DockerSuite) TestRunNamedVolumeCopyImageData(c *check.C) {
 	out, _ := dockerCmd(c, "run", "-v", "foo:/foo", "busybox", "cat", "/foo/hello")
 	c.Assert(strings.TrimSpace(out), check.Equals, "hello")
 }
+
+func (s *DockerSuite) TestRunWithPull(c *check.C) {
+	testRequires(c, Network)
```

Comment:
This test (at least) should have `NotArm` on it as it will download an image that is not for `ARM` arch and thus on arm (build) will fail.